src/libcore/fmt/mod.rs

```diff
@@ -166,7 +166,9 @@ pub struct Formatter<'a> {
 // NB. Argument is essentially an optimized partially applied formatting function,
 // equivalent to `exists T.(&T, fn(&T, &mut Formatter) -> Result`.
 
-enum Void {}
+struct Void {
+    _private: (),
+}
```

Comment:
IOW, this is a good/necessary change long-term, but it shouldn't be *needed* here - if it is, something is wrong with the assumption we're making which is the basis for allowing this PR through.